sqlassignment3.txt

```diff
@@ -0,0 +1,353 @@
+
+Create a string of 10 asterisks (*).
+
+SELECT REPLICATE('*', 10) as stringg;
+
+ Write a query to pad all product codes in a 'Products' table to a length of 10 characters with leading zeros.
+
+update Products
+set ProductCode=REPLICATE('0',10-len(ProductCode))+ProductCode
+
+	Insert the string 'New ' at the beginning of 'York City'.
+
+declare @string1 Varchar(30)='York City'
+declare @string2 Varchar(30)='New'
+declare @string3 Varchar(30)=@string2+' '+@string1
+select @string3
+
+	 From an 'Emails' table, mask the username part of email addresses, showing only the first and last characters.
+
+select * from email
+select left(email,1)+replicate('*',len(left(email,CHARINDEX('@',email)))-3)+right(email,len(email)-CHARINDEX('@',email)+2) from email
+
+	Convert the string 'sql server' to uppercase.
+
+select upper('sql server')
+
+	Write a query to convert all customer names in a 'Customers' table to uppercase.
+
+update customers
+set CustomerName=upper(CustomerName)
+
+	Convert the string 'SQL SERVER' to lowercase.
+
+select lower('SQL SERVER')
+
+	From a 'Products' table, write a query to convert all product descriptions to lowercase.
+
+update Products
+set product_descriptions=lower(product_descriptions)
+
+
+	Remove trailing spaces from the string 'SQL Server    '.
+
+select RTRIM('SQL Server    ')
+
+	Write a query to remove trailing spaces from all email addresses in an 'Employees' table.
+
+update employees
+set email=RTRIM(email) 
+	
+Remove leading spaces from the string '   SQL Server'.
+
+select LTRIM('   SQL Server')
+
+From a 'Comments' table, write a query to remove leading spaces from all comment texts.
+
+update Comments
+set comments=LTRIM(comments) 
+
+	Display the current date in the format 'dd-MM-yyyy'.
+
+declare @date datetime=getdate()
+select format(@date,'dd-MM-yyyy')
+
+	From an 'Orders' table with an 'OrderTotal' column, display the total as a currency with 2 decimal places.
+
+SELECT FORMAT(ordertotal, 'C2', 'en-US') AS CurrencyFormatted from orders;
+
+	Separate the string 'apple,banana,cherry' into individual fruits.
+
+select value as fruits from string_split('apple,banana,cherry',',');
+
+	From a 'Skills' table with a 'SkillList' column containing comma-separated skills, write a query to create a row for each individual skill.
+
+select * from skill cross apply string_split(skill_list,',')
+
+Date and Time Functions
+
+	Write a query to display the current date and time.
+
+select getdate() as datetime
+
+	From an 'Orders' table, find all orders placed in the last 24 hours.
+
+SELECT * FROM Orders WHERE DATEDIFF(HOUR, Order_placed_time, GETDATE()) <= 24;
+
+	Display the current UTC date and time.
+
+select GETUTCDATE() as utcdatetime
+
+	Write a query to show the time difference between local time and UTC time.
+
+SELECT dateDIFF(MINUTE, GETUTCDATE(), GETDATE());
+
+	Convert the current date and time to 'Pacific Standard Time'.
+
+SELECT GETDATE() at time zone 'Pacific standard time' as pacific_std_time
```

Comment:
(https://learn.microsoft.com/en-us/sql/t-sql/queries/at-time-zone-transact-sql?view=sql-server-ver16#b-convert-values-between-different-time-zones)

First we need to specify the time zone of our stored data, then specify the target time zone.

Reply:
SELECT GETDATE() at time zone  'UTC'  at time zone  'Pacific standard time' as pacific_std_time;